### PR TITLE
contracts-bedrock: foundry-ify

### PR DESCRIPTION
**Pull request by tynes**
**Description**

Move the `contracts` directory to `src` and then move the tests into a top level `test` dir so that its
standard foundry style for a repo.

This is going to break any in flight PRs so worth trying to get in relatively quickly, will be no fun to rebase a bunch of PRs.

No semver bumps required because no production source code is changing

### CODE REVIEW

**Review by refcell — COMMENTED** [review]
King, looks good to me, but will await other's review before approving

**Comment by tynes** [comment]
@maurelian asked for this this morning

**Comment by tynes** [comment]
The last thing is the artifacts output path but not changing that as part of this PR as it will require a diff in typescript code

File: packages/contracts-bedrock/foundry.toml [hunk 1]
```diff
@@ -1,8 +1,9 @@
 [profile.default]
-src = 'contracts'
+src = 'src'
```
Comment (maurelian) [inline packages/contracts-bedrock/foundry.toml:2]: probably can delete this?

**Review by maurelian — APPROVED** [review]
No bindings changed, so I checked the diff in the storage layout, everything LGTM.
Thanks for this!

**Pull request merged**